pkg/plugin/processor/builtin/renameField.go

```diff
@@ -0,0 +1,125 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=renameField_paramgen.go renameFieldConfig
+
+package builtin
+
+import (
+	"context"
+	"strings"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"golang.org/x/exp/slices"
+)
+
+type renameField struct {
+	mapping            map[string]string
```

Comment:
Keys (old field reference) in `mapping` are unused. Maybe we can use a slice here, and match references and new field names with indexes? E.g. referenceResolvers[i] is renamed to `newNames[i]`.